tests/test_trimming.py

```diff
@@ -1363,17 +1408,49 @@ def queueData(self, queueCounters):
         yield meta_dict
 
         sai_attr_dict = {
-            "SAI_QUEUE_STAT_TRIM_PACKETS": "0"
+            "SAI_QUEUE_STAT_TRIM_PACKETS": "0",
+            "SAI_QUEUE_STAT_TX_TRIM_PACKETS": "0",
+            "SAI_QUEUE_STAT_DROPPED_TRIM_PACKETS": "0"
         }
 
         trimlogger.info("Reset queue trimming counters: port={}, queue={}".format(self.PORT, self.QUEUE))
         self.dvs_queue.set_queue_counter(queueId, sai_attr_dict)
 
         trimlogger.info("Deinitialize queue data")
 
-    def test_TrimPortStats(self, portData):
+    @pytest.mark.parametrize(
+        "attr, value", [
+            pytest.param("SAI_SWITCH_STAT_TX_TRIM_PACKETS", "1000", id="tx-packet"),
+            pytest.param("SAI_SWITCH_STAT_DROPPED_TRIM_PACKETS", "2000", id="drop-packet")
+        ]
+    )
+    def test_TrimSwitchStats(self, switchData, attr, value):
+        sai_attr_dict = {
+            attr: value
+        }
+
+        trimlogger.info("Update switch counter")
+        self.dvs_switch.set_switch_counter(
+            sai_switch_id=switchData["id"],
+            sai_qualifiers=sai_attr_dict
+        )
+
+        trimlogger.info("Validate switch counter")
+        self.dvs_switch.verify_switch_counter(
+            sai_switch_id=switchData["id"],
+            sai_qualifiers=sai_attr_dict
+        )
+
+    @pytest.mark.parametrize(
+        "attr, value", [
+            pytest.param("SAI_PORT_STAT_TRIM_PACKETS", "1000", id="trim-packet"),
+            pytest.param("SAI_PORT_STAT_TX_TRIM_PACKETS", "2000", id="tx-packet"),
+            pytest.param("SAI_PORT_STAT_DROPPED_TRIM_PACKETS", "3000", id="drop-packet")
+        ]
+    )
```

Comment:
how about clear counters?

Reply:
@kperumalbfn counters clear testing is done as part of `sonic-utilities`. It has nothing to do with the VS tests of the Flex Counter infra. Please pay attention to the fact that counters operate only in a READ but not READ-and-CLEAR mode.